Testing/SystemTests/tests/framework/AbinsTest.py

```diff
@@ -485,3 +485,44 @@ def runTest(self):
     def validate(self):
         self.tolerance = 1e-2
         return self._output_name, self.ref_result
+
+#------------------------------------------------------------------------------
+# Tests for 2D S
+#------------------------------------------------------------------------------
+
+
+class AbinsCRYSTAL2D(systemtesting.MantidSystemTest, HelperTestingClass):
+    """
+    Check Abins2D runs successfully for a typical use case
+    """
+    tolerance = None
+    ref_result = None
+
+    def runTest(self):
+        HelperTestingClass.__init__(self)
+
+        name = "AbinsCRYSTAL2D"
+
+        self.ref_result = name + ".nxs"
+        self.set_ab_initio_program("CRYSTAL")
+        self.set_name(name)
+        self.set_order(QUANTUM_ORDER_TWO)
+        self.set_instrument_name("MARI")
+        params_2d = {'Chopper': 'A',
+                     'ChopperFrequency': str(300),
+                     'IncidentEnergy': str(200),
+                     'EnergyUnits': 'meV'}
+
+        Abins2D(AbInitioProgram="CRYSTAL",
+                VibrationalOrPhononFile="TolueneScratchAbins" + self._extension[self._ab_initio_program],
+                TemperatureInKelvin=self._temperature,
+                Instrument=self._instrument_name,
+                Atoms=self._atoms, SumContributions=self._sum_contributions,
+
+                QuantumOrderEventsNumber=str(self._quantum_order_event), autoconvolution=True,
+                ScaleByCrossSection=self._cross_section_factor, OutputWorkspace=self._output_name,
+                **params_2d)
+
+    def validate(self):
```

Comment:
whats this used for?

Reply:
It's a standard SystemTest, works the same way as the others in this file. https://developer.mantidproject.org/SystemTests.html#writing-a-test

HelperTestingClass sets `self._output_name`, the test framework calls `runTest()` for the calc and compares the workspaces returned by `validate()`.